os/user/getent_usefgetent.go

```diff
@@ -0,0 +1,86 @@
+// +build aix dragonfly freebsd !android,linux netbsd openbsd solaris !darwin
+// +build cgo,!osusergo
+
+package user
+
+/*
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <grp.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+
+static void resetErrno(){
+	errno = 0;
+}
+*/
+import "C"
+import "unsafe"
+
+// usefgetpwent is used for testing. Sets getpwent to use fgetpwent.
+func (u *iterateUsersHelper) usefgetpwent(fileName string) error {
+	fname := C.CString(fileName)
+	defer C.free(unsafe.Pointer(fname))
+	mode := C.CString("r")
+	defer C.free(unsafe.Pointer(mode))
+
+	var fp *C.FILE
+	C.resetErrno()
+	fp, err := C.fopen(fname, mode)
```

Comment:
I am confused on how this works. C.fopenP() returns a single value; how does it work with 2?

Reply:
As per cgo documentation, we can use multiple assignment when calling c functions in cgo. Second argument is automatically set to C.errno. https://pkg.go.dev/cmd/cgo#:~:text=Any%20C%20function%20(even%20void%20functions)%20may%20be%20called%20in%20a%20multiple%20assignment%20context%20to%20retrieve%20both%20the%20return%20value%20(if%20any)%20and%20the%20C%20errno%20variable%20as%20an%20error%20(use%20_%20to%20skip%20the%20result%20value%20if%20the%20function%20returns%20void).%20For%20example%3A